kobotoolbox/kpi
# refactor(organizations): code cleanup and patch data cleanup in account settings and tos view TASK-1292

### 👀 Preview steps
1. ℹ️ have an account and a project
2. Go into account settings view
3. Change information on some field(s)
4. Save the changes
5. Check in the network tab the call to the `/me` endpoint. The payload should only contain the changed fields.
6. Enable a TOS view by following [this document](https://www.notion.so/kobotoolbox/Setting-up-TOS-1447e515f65480f8ab34f69d42d806f8)
7. In the TOS view, fill in the information and save (you may want to `preserve log` before saving data)
8. Check the network tab. Payload should contain only the changed field.

### 💭 Notes
We're refactoring accountDetails and TOS view on the way of implementing some changes on displayed fields:
- Removed the "global" observer from mob-x which was causing linting issues in favor of using the `useSession` hook created at https://github.com/kobotoolbox/kpi/pull/5303
- Cleaned up the code
- Changed in a way that only changed fields are sent to the `PATCH` request to update data
